aea/mail/base.py

```diff
@@ -311,13 +314,54 @@ class Envelope:
 
     default_serializer = DefaultEnvelopeSerializer()
 
+    @staticmethod
+    def _get_protocol_specification_id_by_the_protocol_id(
+        protocol_id: PublicId,
+    ) -> PublicId:
+        """
+        Get protocol specification id  from protocol id.
+
+        If not protocol was registered with given specification id, returns protocol specification id  itself.
+
+        :param protocol_id: PublicId
+
+        :return: PublicId
+        """
+        return (
+            ProtocolSpecificationsRegistry.get_specification_id_by_protocol_id(
+                protocol_id
+            )
+            or protocol_id
+        )
+
+    @staticmethod
+    def _get_protocol_id_by_protocol_specification_id(
+        protocol_specification_id: PublicId,
+    ) -> PublicId:
+        """
+        Get protocol id from specification id.
+
+        If not protocol was registered with given specification id, returns protocol_id itself.
+
+        :param protocol_specification_id: PublicId
+
+        :return: PublicId
+        """
+        return (
+            ProtocolSpecificationsRegistry.get_protocol_id_by_specification_id(
+                protocol_specification_id
+            )
+            or protocol_specification_id
+        )
+
```

Comment:
Why don't put these static methods as _public_ functions of the module? It is not tight to the `Envelope` class internals, and it can be reused from other parts of the framework.

Reply:
the main idea was to encapsulated protocol specification id processing in the Envelope and Serializer, so other parts of the system will know nothing about.

We can make methods public if it will be used outside of Envelope or tests.